packages/flutter_tools/lib/src/android/migrations/gradle_java_version_conflict_migration.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../base/file_system.dart';
+import '../../base/project_migrator.dart';
+import '../../base/version.dart';
+import '../../project.dart';
+import '../android_studio.dart';
+
+final Version _androidStudioFlamingo = Version(2022, 2, 0);
+final RegExp _gradleVersionMatch = RegExp(
+  r'\s*distributionUrl=https\\://services\.gradle\.org/distributions/gradle-((?:\d|\.)+)-(?:all|bin)\.zip');
```

Comment:
```suggestion
  r'^\s*distributionUrl\s*=\s*https\\://services\.gradle\.org/distributions/gradle-((?:\d|\.)+)-(?:all|bin)\.zip');
```

The leading `^` is important so that this doesn't match commented-out lines.

And then @reidbaker found in one of the threads on #123916 (GitHub makes it hard to find which one) that there can be whitespace around the `=`, too. No known tool generates it that way, but it gets accepted if someone put it there by hand.

Reply:
Good catch regarding commented lines! Changed to suggested